circuits/transfer/src/gadgets.rs

```diff
@@ -22,3 +23,43 @@ pub fn commitment(
 
     composer.point_addition_gate(p1, p2)
 }
+
+/// Conditionally select a point against identity
+///
+/// Returns:
+/// a, if flag == 1 ^ b == identity
+/// b, if flag == 0 ^ a == identity
+///
+/// Fail the circuit otherwise
+pub fn identity_select_point(
+    composer: &mut StandardComposer,
+    flag: Variable,
+    identity: Point,
+    a: Point,
+    b: Point,
+) -> Point {
+    let flag = composer.boolean_gate(flag);
+
+    let selected = composer.conditional_point_select(a, b, flag);
+    let discarded = composer.conditional_point_select(b, a, flag);
+
+    composer.assert_equal_point(discarded, identity);
+
+    selected
+}
```

Comment:
Since this is more generic, we could move that to `plonk_gadgets`. Is fine to merge now if you want. But file an issue to migrate it at some point.